### PR TITLE
Gh1101 fix localmathalphabets

### PR DESCRIPTION
**Pull request by FrankMittelbach**
# Internal housekeeping

## Status of pull request

 Ready to merge

## Checklist of required changes before merge will be approved
- [x] Test file(s) added
- [x] Version and date string updated in changed source files
- [x] Relevant `\changes` entries in source included
- [x] Relevant `changes.txt` updated
- [???] Rollback provided (if necessary)?
- [x] `ltnewsX.tex` (and/or `latexchanges.tex`) updated

### CODE REVIEW

File: base/doc/ltnews41.tex [hunk 1]
```diff
@@ -229,11 +229,28 @@ \subsection{Logging text command and symbol declarations}
...
+occasionally. The idea was to avoid allocating all math alphabets
+globally, but instead allow a number of them (defined by counter
+\texttt{localmathalpabets}) to vary from one formula to the next. This
+way different formulas can make use of different alphabets and chances
+are much higher that the processing a complex the document succeeds.
+See~\cite{41:ltnews34} for details.
+
+Unfortunately, the approach taken failed in some cases of nested
+formulas with the result that the wrong symbol glyphs were used
+(rather than, as before, your \LaTeX{} run stopped with a fatal error
```
Comment (davidcarlisle) [inline base/doc/ltnews41.tex:247]: In some cases the original allocation would have worked (and the issue could be avoided by setting localmathalphabets to 0 if you get close enough to trigger the local mathalphabet mechanism without going over 16 fam . You could perhaps simply delete the parenthetic remark..

File: base/ltfssbas.dtx [hunk 2]
```diff
@@ -1184,18 +1184,25 @@
...
          {\@nomath\mathversion
           \expandafter\ifx\csname mv@#1\endcsname\relax
           \@latex@error{Math version `#1' is not defined}\@eha\else
 %    \end{macrocode}
-%  If there has been a frozen math version reset locally. See GH 1028.
+%    If there has been a frozen math version reset unconditionally to
+%    it, so that in case of bold inside normal math we always restart.
+%    This probably fails if you have several levels of nesting with
+%    different math versions so perhaps not a good idea. Maybe in
+%    addition one would need to look at the \cs{@math@level} and do
```
Comment (davidcarlisle) [inline base/ltfssbas.dtx:1199]: I think in the code below you are now testing math@level =1 so "Maybe is addition" not really applying.

File: base/ltfssdcl.dtx [hunk 3]
```diff
@@ -714,33 +714,67 @@
 %    \begin{macrocode}
   \cs_gset:cpn{mv@#1@reset}
      {
+%    \end{macrocode}
+%    If we are back at top-level, or more precisely outside of any
+%    (nested) math, we may have to reset the math version back to its
+%    frozen version, if that exist and has less alphabets allocated.
```
Comment (davidcarlisle) [inline base/ltfssdcl.dtx:720]:
```suggestion
%    frozen version, if that exists and has fewer alphabets allocated.
```

**Review by davidcarlisle — CHANGES_REQUESTED** [review]
Code looks good some more or less cosmetic doc comments just the one about testing nesting level I think probably should change, the others just suggestions.

File: base/ltfssbas.dtx (hunk 2 above)
Comment (FrankMittelbach) [reply base/ltfssbas.dtx:1199]: yeah a leftover not updated after improving the implementation

**Review by davidcarlisle — APPROVED** [review]

**Pull request merged**